test/NuGet.Core.Tests/NuGet.Build.Tasks.Console.Test/MSBuildStaticGraphRestoreTests.cs

```diff
@@ -125,8 +127,34 @@ public void GetPackageDownloads_WhenNotExactVersion_ThrowsException(string versi
                 {
                     var _ = MSBuildStaticGraphRestore.GetPackageDownloads(project).ToList();
                 };
+                string expectedMessage = string.Format(Strings.Error_PackageDownload_OnlyExactVersionsAreAllowed, packageName, expected ?? version);
+                act.Should().Throw<ArgumentException>().WithMessage(expectedMessage);
+            }
+        }
+
+        [Fact]
+        public void GetPackageDownloads_NoVersion_ThrowsException()
+        {
+            string packageName = "PackageA";
+            using (var testDirectory = TestDirectory.Create())
+            {
```

Comment:
Consider dropping the block, which means less indentation...

```suggestion
            using var testDirectory = TestDirectory.Create();
```